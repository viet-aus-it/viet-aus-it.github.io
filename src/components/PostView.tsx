/* eslint-disable react/no-danger */

import { Link } from 'gatsby';
import { rhythm } from '../utils/typography';
import formatPublishedDate from '../utils/formatPublishedDate';

interface PostViewProp {
  node: ContentfulPost;
}

function PostView({ node }: PostViewProp) {
  const { title, description } = node;
  const publishedDate = formatPublishedDate(node.publishDate);

  return (
    <article>
      <header>
        <h3
          style={{
            marginBottom: rhythm(1 / 4),
          }}
        >
          <Link style={{ boxShadow: `none` }} to={node.slug}>
            {title}
          </Link>
        </h3>
        <small>{publishedDate}</small>
      </header>
      <section
        dangerouslySetInnerHTML={{
          __html: description.childMarkdownRemark.html || '<p></p>',
        }}
      />
    </article>
  );
}

export default PostView;
