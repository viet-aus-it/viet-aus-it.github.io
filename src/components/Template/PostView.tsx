/* eslint-disable react/no-danger */

import { Link } from 'gatsby';
import { rhythm } from '../../utils/typography';
import formatPublishedDate from '../../utils/formatPublishedDate';

function PostView({ node }: { node: ContentfulPost }) {
  const {
    title,
    description: { description },
  } = node;
  const postContent = description;
  const publishedDate = formatPublishedDate(node.publishDate);

  return (
    <article>
      <header>
        <h3
          style={{
            marginBottom: rhythm(1 / 4),
          }}
        >
          <Link style={{ boxShadow: `none` }} to={node.slug}>
            {title}
          </Link>
        </h3>
        <small>{publishedDate}</small>
      </header>
      <section>
        <p
          dangerouslySetInnerHTML={{
            __html: postContent,
          }}
        />
      </section>
    </article>
  );
}

export default PostView;
