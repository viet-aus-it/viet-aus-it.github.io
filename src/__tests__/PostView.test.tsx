import { render } from '@testing-library/react';
import PostView from '../components/Template/PostView';

describe('Post View', () => {
  it('should render "proper" post nodes', () => {
    const fullNode = {
      fields: {
        slug: 'Test Post',
      },
      frontmatter: {
        title: 'Test Post',
        date: '27/04/2020',
        description: '<p>This is just a test post</p>',
      },
      except: '<p>This is just a test post</p>',
    };

    const tree = render(<PostView node={fullNode} />);
    expect(tree).toMatchSnapshot();
  });

  it('should render post nodes with missing frontmatter details', () => {
    const withoutFrontmatterTitle = {
      fields: {
        slug: 'Test Post',
      },
      frontmatter: {
        date: '27/04/2020',
        description: '<p>This is just a test post</p>',
      },
      except: '<p>This is just a test post</p>',
    };
    const tree2 = render(<PostView node={withoutFrontmatterTitle} />);
    expect(tree2).toMatchSnapshot();

    const withoutFrontmatterDesc = {
      fields: {
        slug: 'Test Post',
      },
      frontmatter: {
        title: 'Test Post',
        date: '27/04/2020',
      },
      except: '<p>This is just a test post</p>',
    };
    const tree3 = render(<PostView node={withoutFrontmatterDesc} />);
    expect(tree3).toMatchSnapshot();

    const withoutFrontmatterTitleAndDesc = {
      fields: {
        slug: 'Test Post',
      },
      frontmatter: {
        date: '27/04/2020',
      },
      except: '<p>This is just a test post</p>',
    };
    const tree4 = render(<PostView node={withoutFrontmatterTitleAndDesc} />);
    expect(tree4).toMatchSnapshot();
  });
});
