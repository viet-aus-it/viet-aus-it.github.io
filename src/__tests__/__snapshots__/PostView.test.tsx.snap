// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post View should render "proper" post nodes 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <article>
        <header>
          <h3
            style="margin-bottom: 0.4375rem;"
          >
            <a
              href="Test Post"
              style="box-shadow: none;"
            >
              Test Post
            </a>
          </h3>
          <small>
            Mar 27, 2020
          </small>
        </header>
        <section>
          <p>
            Test Description
          </p>
        </section>
      </article>
    </div>
  </body>,
  "container": <div>
    <article>
      <header>
        <h3
          style="margin-bottom: 0.4375rem;"
        >
          <a
            href="Test Post"
            style="box-shadow: none;"
          >
            Test Post
          </a>
        </h3>
        <small>
          Mar 27, 2020
        </small>
      </header>
      <section>
        <p>
          Test Description
        </p>
      </section>
    </article>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
